internal/appsec/listener/sharedsec/shared.go

```diff
@@ -38,6 +37,20 @@ func RunWAF(wafCtx *waf.Context, values waf.RunAddressData) waf.Result {
 	return result
 }
 
+func MakeWAFRunListener[O dyngo.Operation, T dyngo.ArgOf[O]](events *trace.SecurityEventsHolder, wafCtx *waf.Context, limiter limiter.Limiter, toRunAddressData func(T) waf.RunAddressData) func(O, T) {
```

Comment:
I'd break this down to improve legibility

```suggestion
func MakeWAFRunListener[O dyngo.Operation, T dyngo.ArgOf[O]](
	events *trace.SecurityEventsHolder,
	wafCtx *waf.Context,
	limiter limiter.Limiter,
	toRunAddressData func(T) waf.RunAddressData,
) func(O, T) {
```